tests/messages/test_pofile.py

```diff
@@ -299,10 +299,37 @@ def test_obsolete_message_with_context(self):
         catalog = pofile.read_po(buf)
         assert len(catalog) == 2
         assert len(catalog.obsolete) == 1
-        message = catalog.obsolete["foo"]
+        message = catalog.obsolete[("foo", "other")]
         assert message.context == 'other'
         assert message.string == 'Voh'
 
+    def test_obsolete_messages_with_context(self):
```

Comment:
Could we add a roundtrip test here – `write_po(ignore_obsolete=False)`/`generate_po(ignore_obsolete=False)` reads `.obsolete`, so we should maybe see that we get the same stuff back out?

Reply:
Added!